### PR TITLE
Test dockerfile builds on push/pull-request

### PR DESCRIPTION
**Pull request by kgrubb**
This pr will build each dockerfile whenever a branch is pushed or a pull request is created, which should help catch any failing builds before they're pushed to the master branch and published.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    name: ${{ matrix.image }}
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Clone
+      uses: actions/checkout@v1
+
+    - name: Build Docker Image
+      run: |
+        docker login -u "${{ secrets.DockerUsername }}" -p "${{ secrets.DockerPassword }}"
```
Comment (btkostner) [inline .github/workflows/test.yml]: Why login here? Considering we are not pushing anything, best to leave this out just for extra security.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        TAG=${IMAGE#*-}
+        cd "$IMAGE" || exit 1
+        # don't build default tags.
+        declare -a arr=("stable" "stable-installer" "unstable" "unstable-installer" "latest")
+        if [[ ! "${arr[@]}" =~ "$IMAGE" ]]; then
+          # rename unstable tags to latest
+          if [ "$TAG" == "unstable" ]; then TAG="latest"; fi
+          if [ "$TAG" == "unstable-installer" ]; then TAG="installer"; fi
+          docker build -t elementary/"$IMAGE_NAME":"$TAG" .
+        fi
```
Comment (btkostner) [inline .github/workflows/test.yml]: Considering we are not actually publishing anything here, we could simply just do `docker build -t "${{ matrix.image }}" -f "${{ matrix.image }} ." and not do all of the rename stuff. (we will still want to ignore the default tags)

**Review by btkostner — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 2 above)
Comment (kgrubb) [reply .github/workflows/test.yml]: i was lazy and copy-pasted. :scream: I've updated the job to just run the builds.

**Comment by btkostner** [comment]
@kgrubb `-f ${{ matrix.image }}/Dockerfile` my bad

**Review by btkostner — APPROVED** [review]

**Pull request merged**